File: BEMLIB_22.10/05_stokes/flow_3x/ell_int.f
      subroutine ell_int (RK,F,E)

c-----------------------------------------
c FDLIB
c
c Copyright by C. Pozrikidis, 1999
c All rights reserved.
c
c This program is to be used only under the
c stipulations of the licensing agreement.
c----------------------------------------

c------------------------------------------------
c This program accompanies the book:
c
c            C. Pozrikidis
c ``Numerical Computation in Science and Engineering''
c       Oxford University Press
c------------------------------------------------

c-------------------------------------------
c  Evaluation of complete elliptic integrals
c  of the first and second kind
c  using a recursion formula
c
c  SYMBOLS:
c  --------
c
c  F:	first kind
c  E:	second kind
c  acc: specified accuracy
c-------------------------------------------

      Implicit Double Precision (a-h,o-z)

      Parameter (acc=0.000000001)

c----------
c constants
c----------

      pi  = 3.14159 265358  D0
      pih = 0.5D0*pi
      
c----------
c launching
c----------

      F  = pih
      E  = 1.0D0
      G  = 1.0D0
      B  = RK
 
c     Do while (abs(D).GT.acc)

  1   Continue

      C = Dsqrt(1.0D0-B*B)
      B = (1.0D0-C)/(1.0D0+C)
      D = F*B
      F = F+D
      G = 0.5D0*G*B
      E = E+G

      If(abs(D).gt.acc) Go to 1

c     End Do

      E = F*(1.0D0-0.5D0*RK*RK*E)

c-----
c Done
c-----
 
      Return
      End

File: BEMLIB_22.10/05_stokes/sgf_ax/ell_int.m
function [F,E]= ell_int (k)

%-----------------------------------------
% FDLIB BEMLIB
%
% Copyright by C. Pozrikidis, 1999
% All rights reserved
%
% This program is to be used only under the
% stipulations of the licensing agreement.
%----------------------------------------

%--------------------------------------------
%  computation of complete elliptic integrals
%  of the first (F) and second (E) kind
%--------------------------------------------

tol=0.0000000000001;

%-----------
% iterations
%-----------

ks = k*k;

F = 0.5*pi;
E = 1.0;
G = 1.0;
B = k;
D = 10*tol;

%pause
%disp('one')

while abs(D)>tol,
  C = sqrt(1.0-B^2);
  B = (1.0-C)/(1.0+C);
  D = F*B;
  F = F+D;
  G = 0.50*G*B;
  E = E+G;
%pause
%disp('two')
end

E = F*(1.0-0.50*ks*E);

%-----
% done
%-----

return
